app/Shell/Environment.php

```diff
@@ -28,35 +26,26 @@ public function isWindowsOs(): bool
         return PHP_OS_FAMILY === 'Windows';
     }
 
-    public function portIsAvailable($port): bool
+    public function portIsAvailable(int $port): bool
     {
-        // E.g. Win/Linux: 127.0.0.1:3306 , macOS: 127.0.0.1.3306
-        $portText = $this->isLinuxOs() ? "\:{$port}\s" : "\.{$port}\s";
+        // To check if the socket is available, we'll attempt to open a socket on the port.
+        // If we cannot open the socket, it means there's nothing running on it, so the
+        // port is available. If we are successful, that means it is already in use.
+
```

Comment:
Drop the line after the comment block.

Also, I like that you made the fun Laravel -style three comments lines.. but unfortunately, I think we need a bit more -- or a bit different -- explanation. What's missing here is context for anyone who doesn't understand sockets. The natural expectation is that if I try to do a thing on a port and it works, that means the port was available, and if it fails, that means the port wasn't available. But this is the opposite. I think a good comment would explain *that*, more than explaining exactly what the following few lines do.

Reply:
I've changed the comments